docs/docs/adapters/edgedb.md

```diff
@@ -0,0 +1,269 @@
+       property session_state -> str;
+       link user -> User {
+            on target delete delete source;
+       };
+       property createdAt -> datetime {
+            default := datetime_current();
+        };
+
+       constraint exclusive on ((.provider, .providerAccountId))
+    }
+
+    type Session {
+        required property sessionToken -> str {
+            constraint exclusive;
+        }
+        required property userId -> str;
+        required property expires -> datetime;
+        link user -> User {
+            on target delete delete source;
+        };
+        property createdAt -> datetime {
+            default := datetime_current();
+        };
+    }
+
+    type VerificationToken {
+        required property identifier -> str;
+        required property token -> str {
+            constraint exclusive;
+        }
+        required property expires -> datetime;
+        property createdAt -> datetime {
+            default := datetime_current();
+        };
+
+        constraint exclusive on ((.identifier, .token))
+    }
+}
+```
+
+### Migrate the database schema
+
+Create a migration
+
+```
+edgedb migration create
+```
+
+Apply the migration
+
+```
+edgedb migrate
+```
+
+To learn more about [EdgeDB migrations](https://www.edgedb.com/docs/intro/migrations#generate-a-migration), check out the [Migrations docs](https://www.edgedb.com/docs/intro/migrations).
+
+### Generate the query builder
+
+```npm2yarn2pnpm
+npx @edgedb/generate edgeql-js
+```
+
+This will generate the [query builder](https://www.edgedb.com/docs/clients/js/querybuilder) so that you can write fully typed EdgeQL queries with TypeScript in a code-first way.
+
+For example
+
+```ts
+const query = e.select(e.User, () => ({
+        id: true,
+        email: true,
+        emailVerified: true,
+        name: true,
+        image: true,
+        filter_single: { email: 'johndoe@example.com' },
+      }));
+
+return await query.run(client);
+
+// Return type:
+// {
+//     id: string;
+//     email: string;
+//     emailVerified: Date | null;
+//     image: string | null;
+//     name: string | null;
+// } | null
+
+```
+
+
+## Deploying
+
+### Deploy EdgeDB
+
+First deploy an EdgeDB instance on your preferred cloud provider:
+
+[AWS](https://www.edgedb.com/docs/guides/deployment/aws_aurora_ecs)
+
+[Google Cloud](https://www.edgedb.com/docs/guides/deployment/gcp)
+
+[Azure](https://www.edgedb.com/docs/guides/deployment/azure_flexibleserver)
+
+[DigitalOcean](https://www.edgedb.com/docs/guides/deployment/digitalocean)
+
+[Fly.io](https://www.edgedb.com/docs/guides/deployment/fly_io)
+
+[Docker](https://www.edgedb.com/docs/guides/deployment/docker) (cloud-agnostic)
+
+### Find your instance’s DSN
+
+The DSN is also known as a connection string. It will have the format `edgedb://username:password@hostname:port`. The exact instructions for this depend on which cloud you are deploying to. 
+
+### Set an environment variable
```

Comment:
Might be worth noting that the exact mechanism for setting this variable will depend on the hosting platform...e.g. with Vercel you set this through the dashboard.